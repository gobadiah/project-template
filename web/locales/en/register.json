{
  "title": "Register",
  "Create your account": "Create your account",
  "Signin": "Signin"
}
