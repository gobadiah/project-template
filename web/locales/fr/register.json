{
  "title": "Créer un compte",
  "Create your account": "Créer un compte",
  "Signin": "Se connecter"
}
