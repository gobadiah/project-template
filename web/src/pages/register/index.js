import React from 'react';

import { Page } from '~/components/base';
import hoc from '~/hoc';

import { RegisterForm } from '~/auth';

class Register extends Page {
  render() {
    const { register } = this.props;
    return (
      <RegisterForm onSubmit={register} />
    );
  }
}

export default hoc('register')(Register);
