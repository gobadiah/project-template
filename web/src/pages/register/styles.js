import { css } from 'react-emotion';

export const containerClassName = css`
  width: 640px;
  height: 1106px;
`;
