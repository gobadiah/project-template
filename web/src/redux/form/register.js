import React from 'react';

import { reduxForm } from 'redux-form';

import {
  BirthdayField,
  EmailField,
  FirstNameField,
  LastNameField,
  PasswordField,
} from './fields';
import { BaseForm } from './utils';

class RegisterForm extends BaseForm {
  getFields = () => [
    <EmailField key='email' />,
    <PasswordField
      key='password'
      autoComplete='new-password'
    />,
    <PasswordField
      key='confirm_password'
      name='confirm_password'
      label='Password confirmation'
      autoComplete='new-password'
    />,
    <FirstNameField key='first_name' />,
    <LastNameField key='last_name' />,
    <BirthdayField key='birthday' />,
  ];
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
