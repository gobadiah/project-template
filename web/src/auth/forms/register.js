import { css } from 'react-emotion';
import { reduxForm } from 'redux-form';
import React from 'react';

import {
  BirthdayField,
  EmailField,
  FemaleField,
  FirstNameField,
  LastNameField,
  LeftHandedField,
  MaleField,
  PasswordField,
  RankingField,
  RightHandedField,
  TennisClubField,
} from '~/components/forms/fields';
import { Flex } from '~/styles';
import { Form } from '~/components/forms';
import { PureComponent } from '~/components/base';
import { RoundButton } from '~/components';

class RegisterForm extends PureComponent {
  render() {
    const { t } = this.context;

    return (
      <Form {...this.props}>
        <Flex>
          <FirstNameField />
          <LastNameField />
        </Flex>
        <EmailField />
        <Flex>
          <PasswordField autoComplete='new-password' />
          <PasswordField
            name='confirm_password'
            label='Password confirmation'
            autoComplete='new-password'
          />
        </Flex>
        <Flex>
          <BirthdayField />
          <RankingField />
        </Flex>
        <TennisClubField />
        <Flex css='justify-content: space-between'>
          <MaleField inputClassName={css`width: 282px`} />
          <FemaleField inputClassName={css`width: 282px`} />
        </Flex>
        <Flex css='justify-content: space-between'>
          <LeftHandedField inputClassName={css`width: 282px`} />
          <RightHandedField inputClassName={css`width: 282px`} />
        </Flex>
        <RoundButton text={t('I create my account')} className={css`width: 100%`} />
      </Form>
    );
  }
}

export { RegisterForm as PlainRegisterForm };

export default reduxForm({
  form: 'RegisterForm',
})(RegisterForm);
