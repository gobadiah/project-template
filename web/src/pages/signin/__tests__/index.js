import 'jsdom-global/register';

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SignIn from '..';

describe('SignIn', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<SignIn />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shallow rendering should contain SignInForm', () => {
    const wrapper = shallow(<SignIn />);
    wrapper.contains('SignInForm');
  });
});
