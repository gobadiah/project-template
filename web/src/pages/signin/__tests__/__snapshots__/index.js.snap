// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="emotion-0 emotion-1"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        autoComplete="email"
        className={undefined}
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder={undefined}
        type="email"
        value=""
      />
    </div>
    <div
      className="emotion-0 emotion-1"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        autoComplete="current-password"
        className={undefined}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        placeholder={undefined}
        type="password"
        value=""
      />
    </div>
    <input
      type="submit"
      value="Sign in"
    />
  </form>
  <div />
</div>
`;
